src/TConsole.cpp

```diff
@@ -2594,3 +2619,206 @@ void TConsole::slot_reloadMap(QList<QString> profilesList)
 
     pHost->postMessage(outcomeMsg);
 }
+
+QPair<bool, QString> TConsole::addSingleWordToSet(const QString& word)
+{
+    QString errMsg = QStringLiteral("the word \"%1\" already seems to be in the user dictionary");
+    QPair<bool, QString> result = qMakePair(false, QString());
+    if (!mEnableUserDictionary) {
+        return qMakePair(false, QLatin1String("a user dictionary is not enable for this profile"));
+    }
+
+    if (!mUseSharedDictionary) {
+        // The return value from this function is unclear - it does not seems to
+        // indicate anything useful
+        Hunspell_add(mpHunspell_profile, word.toUtf8().constData());
+        if (!mWordSet_profile.contains(word)) {
+            mWordSet_profile.insert(word);
+            qDebug().noquote().nospace() << "TConsole::addSingleWordToSet(\"" << word << "\") INFO - word added to profile mWordSet.";
+            result.first = true;
+        } else {
+            result.second = errMsg.arg(word);
+        }
+
+    } else {
+        auto pMudlet = mudlet::self();
+        QPair<bool, bool> sharedDictionaryResult = pMudlet->addSingleWordToSet(word);
+        while (!sharedDictionaryResult.first) {
+            qDebug() << "TConsole::addSingleWordToSet(...) ALERT - failed to get a write lock to access mWordSet_shared and loaded shared hunspell dictionary, retrying...";
+            sharedDictionaryResult = pMudlet->addSingleWordToSet(word);
+        }
+
+        if (sharedDictionaryResult.second) {
+            // Successfully added word:
+            result.first = true;
+        } else {
+            // Word already present
+            result.second = errMsg.arg(word);
+        }
+    }
+
+    return result;
+}
+
+QPair<bool, QString> TConsole::removeSingleWordFromSet(const QString& word)
```

Comment:
It ***is*** a `QSet<QString>` - for speed of checking for inclusion/so that there is no need to avoid the duplication that a `QStringList` (or `QList<QString>`) would require - so I felt calling it `removeSingleWordFromList` was *less* intuitive/consistent.

The name also reflected that there was a (now banished) method to add all the words in a line (that were not in the main dictionary AND were above a certain number of graphemes long AND {optionally} did not contain numbers) of text to the same container.

OTOH It does make sense now to remove the `single` from the `add...` and `remove...` methods.

Reply:
Have removed the single in next commit.